hypernets_processor/main/sequence_processor_main.py

```diff
@@ -83,10 +95,17 @@ def get_target_sequences(context, to_archive):
 
         raw_products = [os.path.basename(raw_path) for raw_path in raw_paths]
         raw_products = list(set(raw_products) - set(complete_products))
-        raw_paths = [
-            os.path.join(directory, raw_product) for raw_product in raw_products
+
+        raw_products_datetimes = [parse_sequence_path(raw_product)["datetime"] for raw_product in raw_products
         ]
 
+        raw_paths = []
+        for i in range(len(raw_products)):
+            if np.abs(raw_products_datetimes[i]-datetime.datetime.now())<datetime.timedelta(minutes=context.get_config_value("delay_min")):
+                print("%s is not processed yet due to not having reached required delay (%s minutes)"%(raw_products[i],context.get_config_value("delay_min")))
+            else:
+                raw_paths.append(os.path.join(directory, raw_products[i]))
```

Comment:
This doesnt work with YYYY/MM/DD/SEQ* directory structure because there are many different directories, not just the first one given by directory = os.path.dirname(raw_paths[0])